### PR TITLE
Maverick L (laser guided AGM)

### PR DESCRIPTION
**Pull request by xrufix**
**When merged this pull request will:**
- Add AGM-65 Maverick L type laser guided missiles.
-- Utilizes ACE Advanced Missile Guidance.
-- Uses the HUD which is also used in ace_hellfire.
-- Pylon magazines included - available to all racks on which vanilla Macers can be used.
-- DIR attack profile.

- Rename IR-guided Maverick to "AGM-65 Maverick G".
- Add names of pylon magazines of Macer/Maverick to ace_realisticnames. 

My aim was to provide a laser guided AGM as an addition to vanilla's IR-guided missiles.
The Maverick L is such a weapon, being a somewhat modernized version of the Maverick E/E2 used by US Navy since 2012.

### CODE REVIEW

File: addons/maverick/functions/fnc_attackProfile.sqf [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+ * [[1,2,3], [], []] call ace_maverick_fnc_attackProfile;
+ *
+ * Public: No
+ */
+#define DEBUG_MODE_FULL
+#include "script_component.hpp"
+
+params ["_seekerTargetPos", "_args"];
+_args params ["_firedEH"];
+_firedEH params ["_shooter","","","","","","_projectile"];
```
Comment (Neviothr) [inline addons/maverick/functions/fnc_attackProfile.sqf]: Space after the commas.

**Comment by BaerMitUmlaut** [comment]
It's there anything special about the attack profile? Because there is already a linear one in missile guidance.

**Comment by xrufix** [comment]
The attack profile is a 90% copy of the linear profile. I only removed the upwards arc at the start to prevent missiles climbing after being fired in low altitude flight.

I just checked the differences again, though, and found it will only make a difference in altitudes lower than 5m AGL, so I might as well remove it and use AMGs LIN profile instead.

I have yet to research if Mavericks have a top-attack-mode or similar alternative attack profiles in real-life. Could not find any information about this yet.

**Comment by PabstMirror** [comment]
Looks pretty good to me,

To avoid code duplication I'm thinking we should make the showHud code more generic and just move it to the laser component so that it could work for any weapon system.
There will be some changes to it in #5305 and #5365, so don't worry about doing anything yet.

**Comment by tecak** [comment]
Please make a attackProfile ace_missileguidance for gbu bombs

**Comment by BaerMitUmlaut** [comment]
@tecak This is not the place to do feature requests.

**Comment by xrufix** [comment]
With #5305 and #5365 merged, is there any news on moving the showHud code?

**Comment by PabstMirror** [comment]
Sorry for the delay xrufix, I switched this to use the showHud that's now in ace_laser

**Comment by PabstMirror** [comment]
Self designation is a little tricky due to plane's speed and view distance limits in arma
but is possible (tested in the a-10, using the targeting pod)

**Comment by jonpas** [comment]
> Self designation is a little tricky due to plane's speed and view distance limits in arma

FoV view distance can help a lot I found out, so when you zoom in with the targeting pod view distance increases automatically.

**Pull request merged**